import_export/resources.py

```diff
@@ -966,10 +966,22 @@ def iter_queryset(self, queryset):
         else:
             yield from queryset.iterator(chunk_size=self.get_chunk_size())
 
-    def export(self, queryset=None, *args, **kwargs):
+    def export(self, *args, queryset=None, **kwargs):
         """
         Exports a resource.
-        """
+        :returns: Dataset object.
+        """
+        if len(args) == 1 and (isinstance(args[0], QuerySet) or isinstance(args[0], list)):
+            # issue 1565: definition of export() was incorrect
+            # if queryset is being passed, it must be as the first arg or named parameter
+            # this logic is included for backwards compatibility:
+            # if the method is being called without a named parameter, add a warning
+            # this check should be removed in a future release
+            warnings.warn(
+                "'queryset' must be supplied as a named parameter",
+                category=DeprecationWarning
+            )
+            queryset = args[0]
```

Comment:
Nice, this is very thoughtful! I actually might have some code fixes to do here in my team's stuff.

Reply:
Yes, I have tried to make this change backwards compatible but there may be edge cases which I cannot foresee.  It is broken as it is so we have to change it.